### ISSUE TITLE
A compilation from sources v0.5.1-beta-rc1 -> v0.5.1-beta-rc2 failed

### ISSUE THREAD

**Issue by Perlover** [lightningnetwork/lnd#2196]
### Background

An update from sources from v0.5.1-beta-rc1 to v0.5.1-beta-rc2 failed 

### Your environment

* version of `lnd` - v0.5.1-beta-rc1 (tag)

### Steps to reproduce

1. `cd $GOPATH/src/github.com/lightningnetwork/lnd`
There should be sources of v0.5.1-beta-rc1

2. `git pull; go checkout v0.5.1-beta-rc2`

3. `make clean && make`

### Expected behaviour

Should be compiled

### Actual behaviour

I got:

    $ make
    \033[0;32m Compiling dependencies.\033[0m
    dep ensure -v
    (1/43) Wrote gopkg.in/macaroon.v2@bed2a428da6e56d950bed5b41fcbae3141e5b0d0
    (2/43) Wrote github.com/NebulousLabs/go-upnp@29b680b06c82d044ebea91bf3069038eb562df2a
    (3/43) Wrote github.com/btcsuite/websocket@master
    (4/43) Wrote github.com/btcsuite/go-socks@master
    (5/43) Wrote github.com/btcsuite/fastsha256@master
    (6/43) Wrote github.com/btcsuite/btclog@84c8d2346e9fc8c7b947e243b9c24e6df9fd206a
    (7/43) Wrote github.com/jessevdk/go-flags@f88afde2fa19a30cf50ba4b05b3d13bc6bae3079
    (8/43) Wrote github.com/davecgh/go-spew@8991bc29aa16c548c550c7ff78260e27b9ab7c73
    (9/43) Wrote git.schwanenlied.me/yawning/bsaes.git@master
    (10/43) Wrote github.com/grpc-ecosystem/grpc-gateway@f2862b476edcef83412c7af8687c9cd8e4097c0f
    (11/43) Wrote github.com/aead/chacha20@master
    (12/43) Wrote github.com/coreos/bbolt@4f5275f4ebbf6fe7cb772de987fa96ee674460a7
    (13/43) Wrote github.com/Yawning/aez@4dad034d9db2caec23fb8f69b9160ae16f8d46a3
    (14/43) Wrote github.com/aead/siphash@83563a290f60225eb120d724600b9690c3fb536f
    (15/43) Wrote github.com/NebulousLabs/fastrand@master
    (16/43) Wrote github.com/jackpal/gateway@v1.0.4
    (17/43) Wrote github.com/golang/protobuf@bbd03ef6da3a115852eaf24c8a1c46aeb39aa175
    (18/43) Wrote github.com/rogpeppe/fastuuid@master
    (19/43) Wrote github.com/btcsuite/btcutil@ab6388e0c60ae4834a1f57511e20c17b5f78be4b
    (20/43) Wrote github.com/btcsuite/btcd@79e00513b1011888b1e675157ab89f527f901cae
    (21/43) Wrote github.com/btcsuite/golangcrypto@master
    (22/43) Wrote github.com/jrick/logrotate@a93b200c26cbae3bb09dd0dc2c7c7fe1468a034a
    (23/43) Wrote github.com/lightningnetwork/lightning-onion@ac4d9da8f1d67c95f1fafdc65e1a4902d6f5a940
    (24/43) Wrote github.com/ltcsuite/ltcd@cdab10132e8c6e4a3ffd112dba54791946d28906
    (25/43) Wrote github.com/miekg/dns@79bfde677fa81ff8d27c4330c35bda075d360641
    (26/43) Wrote github.com/lightninglabs/gozmq@master
    (27/43) Wrote github.com/tv42/zbase32@501572607d0273fc75b3b261fa4904d63f6ffa0e
    (28/43) Wrote github.com/lightninglabs/neutrino@9a42f7df21be82a69f04caa83bce4034dca72764
    (29/43) Wrote github.com/urfave/cli@1efa31f08b9333f1bd4882d61f9d668a70cd902e
    (30/43) Wrote golang.org/x/crypto@614d502a4dac94afa3a6ce146bd1736da82514c6
    (31/43) Wrote golang.org/x/net@ae89d30ce0c63142b652837da33d782e2b0a9b25
    (32/43) Wrote gopkg.in/errgo.v1@b20caedf0710d0988e92b5f2d76843ad1f231f2d
    (33/43) Wrote golang.org/x/time@fbb02b2291d28baffd63558aa44b4b56f178d650
    (34/43) Wrote github.com/jackpal/go-nat-pmp@28a68d0c24adce1da43f8df6a57340909ecd7fdd
    (35/43) Wrote github.com/go-errors/errors@a6af135bd4e28680facf08a3d206b454abc877a4
    (36/43) Wrote google.golang.org/genproto@df60624c1e9b9d2973e889c7a1cff73155da81c4
    (37/43) Failed to write github.com/btcsuite/btcwallet@55c7c63993216f15d57f0118e86e3fe832e37f15
    (38/43) Failed to write golang.org/x/text@v0.3.0
    (39/43) Failed to write google.golang.org/grpc@8dea3dc473e90c8179e519d91302d0597c0ca1d1
    (40/43) Failed to write github.com/juju/loggo@master
    (41/43) Failed to write gopkg.in/macaroon-bakery.v2@94012773d2874a067572bd16d7d11ae02968b47b
    (42/43) Failed to write golang.org/x/sys@ebe1bf3edb3325c393447059974de898d5133eb8
    (43/43) Failed to write github.com/kkdai/bstream@f391b8402d23024e7c0f624b31267a89998fca95
    grouped write of manifest, lock and vendor: error while writing out vendor tree: failed to write dep tree: failed to export github.com/btcsuite/btcwallet: fatal: failed to unpack tree object 55c7c63993216f15d57f0118e86e3fe832e37f15
    : exit status 128

If i do again: `go checkout v0.5.1-beta-rc2; make clean && make` then compilation to be going OK.

**Comment by Perlover** [comment]
The problem was resolved after like this problem for other project
I did
cd $GOPATH/pkg/dep/sources/https---github.com-btcsuite-btcwallet
git pull
and after problem was gone away

But i think it's wrong behaviour of update. May be `lnd` config build files have bug.

**Comment by vegardengen** [comment]
For me, a clean build solved the original issue. Rescan happened only once more after doing this, then next restart it noticed it had already done so and proceeded with a normal restart.

**Comment by Roasbeef** [comment]
This is a dep issue, remove the vendor directory and it will be resolved.

**Comment by Perlover** [comment]
Can you explain where vendor directory is located? In $GOPATH or in project directory?

**Comment by Roasbeef** [comment]
Project directory, it's produced by dep.

**Comment by Perlover** [comment]
> This is a dep issue, remove the vendor directory and it will be resolved.

Now i manually remove the dir `vendor`
BUT! When i fun `make clean`:

    rm -f ./lnd-debug ./lncli-debug
    rm -f -r ./vendor .vendor-new

So this directory was removed in any case (to see my first post)
So please open ticket again.

**Comment by Perlover** [comment]
I will try to update future version of `lnd` again at other servers and will let to know
May be after pull request https://github.com/lightningnetwork/lnd/pull/2200 the problem will go out... :)
